src/ThreeJsPanel.ts

```diff
@@ -605,6 +616,35 @@ export class ThreeJsPanel {
     return this.renderer.getContext().canvas.height;
   }
 
+  getCameraState(): CameraState {
+    return {
+      position: this.camera.position.toArray(),
+      up: this.camera.up.toArray(),
+      target: this.controls.target.toArray(),
+      orthoScale: this.camera.type === "OrthographicCamera" ? this.controls.scale : undefined,
+      fov: this.camera.type === "PerspectiveCamera" ? this.camera.fov : undefined,
+    };
+  }
+
+  setCameraState(state: Partial<CameraState>) {
+    const currentState = this.getCameraState();
+    // Fill in any missing properties with current state
+    const newState = { ...currentState, ...state };
+
+    this.camera.up.fromArray(newState.up).normalize();
+    this.camera.position.fromArray(newState.position);
+    this.controls.target.fromArray(newState.target);
```

Comment:
do you need to do something like controls.update() to make sure the camera gets the correct rotation? Or maybe should set camera rotation here with a little math.
Also one thing I would be slightly worried about is if the camera is being overridden in an incorrect way (like XY mode being given a perspective camera?).  Is there a way to error-check that?  (Maybe make sure ortho modes are getting ortho cameras, and make sure the direction of the camera is consistent with the ortho mode.  3d mode is more free in terms of direction)